xclim/indicators/atmos/_precip.py

```diff
@@ -247,6 +250,28 @@ class HrPrecip(Hourly):
     parameters={"tas": {"kind": InputKind.VARIABLE}, "phase": "solid"},
 )
 
+standardized_precipitation_index = PrecipWithIndexing(
```

Comment:
I think this is not the best class (`PrecipWithIndexing`). Because it 1 - expects only daily inputs, 2- adds the possibility of subsetting the data temporally _before_ computing the indicator, which would mess with this calculation.

I suggest creating a new class (at the top of the file):
```
class StandardizedIndexes(ResamplingIndicator):
    """Resampling but flexible inputs indicators."""
    src_freq = ["D", "M"]
    context = "hydro"
```
This needs the import of `ResamplingIndicator` from `xclim.core.indicator`.